### DISCUSSION TITLE
Difference between `:manage` and `:admin` permissionset actions

### DISCUSSION THREAD

**Discussion by nbelzer** [solidusio/solidus#4271] [Troubleshooting] [answered]
https://github.com/solidusio/solidus/blob/a6c2ceeb2ee6bff832080ebb11773f7b6e4c2df0/core/lib/spree/permission_sets/user_display.rb#L3-L13

Solidus often uses the `:admin` action on a resource instead of `:manage` (which is [supported out of the box by CanCanCan](https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md#can-actions)) for its permission sets.  What is the difference between the two options?

**Comment by jarednorman** [comment]
I'd have to check, but I think it's just an alias for `:manage`?

**Reply by nbelzer** [reply]
I believe it does something more than that, otherwise there is no need to also include `:read` or `:edit` in the abilities (as they would be covered by `:manage`).

**Comment by forkata** [comment, marked as answer]
From the CanCanCan documentation linked in the OP I see that `:manage` is a special permission that 

> means literally any action, not only CRUD ones

In Solidus `:admin` is a custom defined action which is [checked in the base backend controller](https://github.com/solidusio/solidus/blob/e878076f2ed670d07654ab6293a16588743f2fa6/backend/app/controllers/spree/admin/base_controller.rb#L29) and throughout the rest of the engine controllers when performing actions that require the highest level of permission.

My understanding based on the CanCanCan documentation is that if a user has the `:manage` permission, they will also be able to `:admin` a resource.

**Reply by nbelzer** [reply]
This checks out, thanks for explaining.
